types/rlp_unmarshal.go

```diff
@@ -301,86 +301,131 @@ func (t *Transaction) UnmarshalRLP(input []byte) error {
 
 // UnmarshalRLPFrom unmarshals a Transaction in RLP format
 func (t *Transaction) UnmarshalRLPFrom(p *fastrlp.Parser, v *fastrlp.Value) error {
-	elems, err := v.GetElems()
-	if err != nil {
-		return err
+	if v.Len() < 10 {
+		return fmt.Errorf("incorrect number of elements to decode transaction, expected 10 but found %d", v.Len())
 	}
```

Comment:
Should we cover situation where `Type` isn't set (and assume it is `LegacyTx`)? 
I am not entirely sure should we take care about backward compatibility or not, that's why questions like this arise? :slightly_smiling_face:

Reply:
I've updated marshaling logic so we set tx type in any way now in order to be more accurate

Reply:
Yeah, but I envision situation where we would have some block which is created before having EIP-1559 transactions and such transactions wouldn't have marshaled `Type` field.

You are returning [here](https://github.com/0xPolygon/polygon-edge/pull/1111/files#diff-efc5999399a6ff1082b29a52ce57fd7bfd6ebe4789d686a44c70523d2a61c684R409-R411) `LegacyTx` and it is correct, but also an error (and we shouldn't treat it as error in case of "legacy" transactions). So I'd say we should consider `Type` field optional. However as I said: if we are going to run everything from scratch and not worry about backward compatibility, then we are good to go with this.

Reply:
Good point. According to the clack conv, we have to be backward compatible. Done.